rbi/core/array.rbi

```diff
@@ -483,6 +483,43 @@ class Array < Object
   end
   def -(arg0); end
 
+  # Comparison --- Returns an integer (`-1`, `0`, or `+1`) if this array is less
+  # than, equal to, or greater than `other_ary`.
+  #
+  # Each object in each array is compared (using the <=> operator).
+  #
+  # Arrays are compared in an "element-wise" manner; the first element of `ary`
+  # is compared with the first one of `other_ary` using the <=> operator, then
+  # each of the second elements, etc... As soon as the result of any such
+  # comparison is non zero (i.e. the two corresponding elements are not equal),
+  # that result is returned for the whole array comparison.
+  #
+  # If all the elements are equal, then the result is based on a comparison of
+  # the array lengths. Thus, two arrays are "equal" according to Array#<=> if,
+  # and only if, they have the same length and the value of each element is
+  # equal to the value of the corresponding element in the other array.
+  #
+  # `nil` is returned if the `other_ary` is not an array or if the comparison of
+  # two elements returned `nil`.
+  #
+  # ```ruby
+  # [ "a", "a", "c" ]    <=> [ "a", "b", "c" ]   #=> -1
+  # [ 1, 2, 3, 4, 5, 6 ] <=> [ 1, 2 ]            #=> +1
+  # [ 1, 2 ]             <=> [ 1, :two ]         #=> nil
+  # ```
+  def <=>(_); end
+
+  # Equality --- Two arrays are equal if they contain the same number of
+  # elements and if each element is equal to (according to Object#==) the
+  # corresponding element in `other_ary`.
+  #
+  # ```ruby
+  # [ "a", "c" ]    == [ "a", "c", 7 ]     #=> false
+  # [ "a", "c", 7 ] == [ "a", "c", 7 ]     #=> true
+  # [ "a", "c", 7 ] == [ "a", "d", "f" ]   #=> false
+  # ```
+  def ==(_); end
```

Comment:
Every `def ==(` here that doesn't have a type overrides the `T::Boolean` return type of `BasicObject#==`

Reply:
(there are a handful of `==` methods in this PR that need to be typed or deleted)